compiler/parser/RdrHsSyn.hs

```diff
@@ -3487,6 +3518,48 @@ mkInterfaceImplements templateName conName sharedBinds (L loc implements) =
         , cL loc (ValD noExt val)
         ]
 
+    mkImplementsView :: P [LHsDecl GhcPs]
+    mkImplementsView = do
+      let ValidImplementsDeclBlock iface _ view = implements
+
+      case view of
+        Nothing -> pure []
+        Just (L loc (ValidImplementsMethodDecl _ vimdMatches)) -> do
+          let
+            fullViewName =
+              noLoc $ mkRdrUnqual $ mkVarOcc $
+                "_view_" ++ templateInterfaceName
+            shortViewName = noLoc $ mkRdrUnqual $ mkVarOcc "view"
+            signature = TypeSig noExt [fullViewName] $
+              mkHsWildCardBndrs $ mkHsImplicitBndrs $
+                interfaceViewType `mkAppTy` templateType `mkAppTy` interfaceType
+            args = [asPatRecWild "this" conName]
+            exp =
+              L loc
+                (HsLet noExt
```

Comment:
If I read this correctly, this creates `let view = … in view`. Why do we need the let binding here?

Reply:
> I suspect that it is there in method declarations so that methods can invoke themselves

I’d be surprised, recursion in Daml is generally fairly limited (only top-level definitions).

What I could see happening is that for methods you may want to pattern match on arguments and you do the pattern match as part of the let bindings? But for view methods there is no argument so I don’t see why we need it.

Reply:
Because of the view changes in https://github.com/digital-asset/daml/pull/14456, we definitely don't want to overwrite the definition of `view` inside the body - I'm removing this for now, we'll see if anything breaks in https://github.com/digital-asset/daml/pull/14435

Reply:
Looking back at the PR where I introduced `HsLet` for interface methods [ghc#107:compiler/parser/RdrHsSyn.hs:3478-3494](https://github.com/digital-asset/ghc/pull/107/files#diff-e599e1c5f9a0d22ed9a8427c797d148bedca1d348e5abc9950490372287fc8f8R3478-R3494), where the main objective was to allow multiple clauses in method implementations (while dropping the `let` keyword), I think @dylant-da's reasoning matches my thoughts at the time

> the function is a collection of top level declarations with pattern matches and guards, so we need to generate a let binding to be able to define all of the method clauses together before invoking them

Looking at it now, though, I wonder if `HsLamCase` would have been enough, since that supports multiple clauses and even guards, and `-XLambdaCase` [is enabled by default](https://github.com/digital-asset/daml/blob/main/compiler/damlc/daml-opts/daml-opts/DA/Daml/Options.hs#L249). 

Actually, `HsLamCase` sounds like the only right option, since a potential recursive call would be at the wrong type `template -> method_result` instead of `interface -> method_result`. I can do this in a separate PR, what do you think @cocreature?